modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java

```diff
@@ -568,12 +585,80 @@ public void close() {
 
     @Override
     public CompletableFuture<Void> snapshot(Path snapshotPath) {
-        throw new UnsupportedOperationException();
+        synchronized (mux) {
+            try {
+                Files.createDirectories(snapshotPath);
+
+                Path snapshotFile = snapshotPath.resolve(SimpleInMemoryKeyValueStorageSnapshot.FILE_NAME);
+
+                assertTrue(IgniteUtils.deleteIfExists(snapshotFile), snapshotFile.toString());
+
+                Files.createFile(snapshotFile);
+
+                var revsIdxCopy = new HashMap<Long, Map<byte[], ValueSnapshot>>();
+                revsIdx.forEach((revision, entries) -> {
+                    Map<byte[], ValueSnapshot> entries0 = new HashMap<>();
+                    entries.forEach((keyBytes, value) -> entries0.put(keyBytes, new ValueSnapshot(value)));
+
+                    revsIdxCopy.put(revision, entries0);
+                });
+
+                var snapshot = new SimpleInMemoryKeyValueStorageSnapshot(
+                        Map.copyOf(keysIdx),
+                        Map.copyOf(tsToRevMap),
+                        Map.copyOf(revToTsMap),
+                        revsIdxCopy,
+                        rev,
+                        updCntr,
+                        savedCompactionRevision
+                );
+
+                byte[] snapshotBytes = ByteUtils.toBytes(snapshot);
+
+                Files.write(snapshotFile, snapshotBytes, WRITE);
```

Comment:
I find it strange that you first serialize everything into a byte array before writing into a file, instead of writing into the file directly

Reply:
This is a test implementation, I wouldn't like to bother too much with it.